wordpress-mobile/WordPress-Android
# Add backoff delay

Parent #13328 & #13329

This PR adds a backoff strategy to `GetBackupDownloadStatusUseCase` and `GetRestoreStatusUseCase` when retrying because of a failed fetch status. The strategy implemented starts with the normal delay and ups it by a factor of 2 for each subsequent retry.

I also did a small POC on using flowRetry, but I couldn't figure out how to reset the retryAttempts on a successful fetch in that scenario. I chose to submit this solution and keep plugging away at flow/retry separately.

To test:
- Run the backup download/restore process to end and ensure it still works as expected.

PR submission checklist:
